developer-tools/scaleway-sdk/go-sdk.mdx

```diff
@@ -0,0 +1,218 @@
+tags: devtools cli download-cli scaleway-cli command-line 
+dates:
+  validation: 2023-09-21
+  posted: 2021-09-21
+categories:
+  - devtools
+---
+
+The Scaleway Go SDK is a tool provided by Scaleway for developers to interact with its services programmatically. 
+It allows for the automation of tasks like server management, storage operations, and networking setup through Go scripts, handling the complexities of direct API calls and authentication behind simple Go function calls. 
+This makes it easier to integrate Scaleway's cloud resources into applications or manage them through code. 
+
+
+## Installation of the Go SDK
+
+You can install the SDK by using the following command: 
+
+```
+go get github.com/scaleway/scaleway-sdk-go
+```
+
+<Message type="note">
+  This library is made to work with Go 3.8 but should work on the latest versions too.
+</Message>
+
+## Initialization
+
+You need a pair of access and secret keys to connect to Scaleway API. Check out the documentation to find out [how to retrieve them](/identity-and-access-management/iam/how-to/create-api-keys/).
+
+`scaleway` APIs must be initialized with a `scaleway.Client`.
+
+A minimal setup would look like this:
+
+```go
+package main
+
+import (
+	"fmt"
+
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"github.com/scaleway/scaleway-sdk-go/utils"
+)
+
+func main() {
+
+	// Create a Scaleway client
+	client, err := scw.NewClient(
+		// Get your organization ID at https://console.scaleway.com/organization/settings
+		scw.WithDefaultOrganizationID("SCW_DEFAULT_ORGANIZATION_ID"),
+		// Get your credentials at https://console.scaleway.com/iam/api-keys
+		scw.WithAuth("SCW_ACCESS_KEY", "SCW_SECRET_KEY"),
+		// Get more about our availability zones at https://www.scaleway.com/en/docs/console/my-account/reference-content/products-availability/
+		scw.WithDefaultRegion("SCW_REGION"),
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create SDK objects for Scaleway Instance product
+	instanceApi := instance.NewAPI(client)
+
+	// Call the ListServers method on the Instance SDK
+	response, err := instanceApi.ListServers(&instance.ListServersRequest{
+		Zone: scw.ZoneFrPar1,
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	// Do something with the response...
+	for _, server := range response.Servers {
+		fmt.Println("Server", server.ID, server.Name)
+	}
+
+}
+
+```
+
+## Example
+
+To create a Kubernetes cluster on Scaleway using the Go SDK, you would typically interact with the `scaleway` SDK package. The specific details can change, so you should always refer to the latest documentation. 
+Below is a conceptual example of how you might use the Scaleway Go SDK to create a Kubernetes (Kapsule) cluster with 3 nodes:
+
+```go
+package main
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	scaleway "github.com/scaleway/scaleway-sdk-go/scw"
+	k8s "github.com/scaleway/scaleway-sdk-go/api/k8s/v1"
+	vpc "github.com/scaleway/scaleway-sdk-go/api/vpc/v2"
```

Comment:
I have 1 message per imported module saying:
> no required module provides package github.com/scaleway/scaleway-sdk-go/api/k8s/v1; to add it: go get github.com/scaleway/scaleway-sdk-go/api

Is it normal? I have to manually run 3 separate `go get` commands to retrieve the modules. cc @remyleone

Reply:
It works now for some reason, probably missing GO dependencies